src/App.js

```diff
@@ -1,5 +1,139 @@
+const MissionUtils = require("@woowacourse/mission-utils");
+const inputValidation = require("./validation/inputValidation");
+
 class App {
-  play() {}
+  constructor(inputNum, randomNum, overInputNum) {
+    this.inputNum = inputNum;
+    this.randomNum = randomNum;
+    this.overInputNum = overInputNum;
+  }
+
+  play() {
+    this.userInit();
+    this.gameRoutine();
+  }
+
+  gameRoutine() {
+    this.randomNumber();
+    this.userInput();
+  }
+
+  // userInput logic
+  userInit() {
+    MissionUtils.Console.print("숫자 야구 게임을 시작합니다.");
+  }
+
+  userInput() {
+    MissionUtils.Console.readLine("숫자를 입력해주세요 : ", (input) => {
+      this.inputNum = input;
+      if (!this.checkInputValidation()) {
+        throw new Error("잘못된 값을 입력하셨습니다.");
+      }
+      this.game();
+      if (this.randomNum.join("") == this.inputNum) {
+        this.overMessage();
+        this.overChoice();
+        return;
+      } else this.userInput();
+    });
+  }
+
+  checkInputValidation() {
+    const checkNoOverlap = inputValidation.checkNoOverlap(this.inputNum);
+    const checkOnlyNum = inputValidation.checkOnlyNum(this.inputNum);
+    const checkTreeNum = inputValidation.checkThreeNum(this.inputNum);
+    if (!checkNoOverlap || !checkOnlyNum || !checkTreeNum) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  randomNumber() {
+    this.randomNum = [];
+    for (let rotate = 0; rotate < 3; rotate++) {
+      const random = MissionUtils.Random.pickNumberInRange(1, 9);
+      if (this.randomNum.indexOf(random) === -1) {
+        this.randomNum.push(random);
+      } else {
+        rotate--;
+      }
+    }
+    return this.randomNum;
+  }
+
+  // 종료 후 logic
+  overMessage() {
+    MissionUtils.Console.print("3개의 숫자를 모두 맞히셨습니다! 게임 종료");
+  }
+
+  overChoice() {
+    MissionUtils.Console.readLine(
+      "게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.\n",
+      (input) => {
+        this.overInputNum = input;
+        if (this.overInputNum == 1) {
+          this.gameRoutine();
+        } else if (this.overInputNum == 2) {
+          MissionUtils.Console.close();
+        } else {
+          throw new Error("잘못된 값을 입력하셨습니다.");
+        }
```

Comment:
```suggestion
        if (this.overInputNum !== '1' && this.overInputNum !== '2') {
          throw new Error("잘못된 값을 입력하셨습니다.");
        }
        (this.overInputNum === '1') ? this.gameRoutine() : MissionUtils.Console.close();
```
예외처리를 먼저 해준다면 조건이 2개만 남게되고, 둘 다 한줄짜리 간단한 조건으로 삼항연산자를 사용하면 더욱 간단하게 표현할 수 있을듯 합니다.

Reply:
리뷰 정말 감사드립니다 ☺️
혹시 간단하게 작성하기 위해서 외에 예외처리를 먼저 해주면 실행속도가 빨라지는 등의 장점이 더 있을까요?

Reply:
아마 프로그램이 커진다면 예외처리를 먼저 하는것이 실행속도면에서는 조금 더 느릴거같아요..
대부분의 입력값이 1이나 2를 가정한다면 예외처리를 수행하지 않고 조건을 만족하고 수행할테니까요.
하지만 현재 야구게임 정도의 프로그램에서는 실행속도보다 클린코드를 지향하는것이 더 우선시 되어야 할것같아서 말씀드렸습니다 !